### ISSUE TITLE
Fn::ForEach Throws "E0001 Error Transforming Template: Duplicate Type while doing transformation" When Used in Resources Section

### ISSUE THREAD

**Issue by jbq2** [aws-cloudformation/cfn-lint#3888]
**cfn-lint 1.18.4**

The following snippet shows something similar to what I am attempting to do (hiding exact details of course):
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::LanguageExtensions'
...
Resources:
  Fn::ForEach::Loop:
    - Id
    - - A
      - B
      - C
    - ScheduleGroup${Id}:
      Type: AWS::Scheduler::ScheduleGroup    # <--- complains about this
      Properties:
        Name: !Ref Id
        ...
```
When I run this through `cfn-lint`, the linter is complaining that there is a Duplicate Type while doing transformation, specifically highlighting the `Type` field under `ScheduleGroup${Id}`. I believe I have followed the examples in AWS docs properly, specifically the 'Replicate Multiple Resources` of [this article](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-foreach-example-resource.html). I expect that the for each loop expands to the following template after the transformation is performed:
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::LanguageExtensions'
...
Resources:
  ScheduleGroupA:
    Type: AWS::Scheduler::ScheduleGroup
    Properties:
      Name: A
      ...
  ScheduleGroupB:
    Type: AWS::Scheduler::ScheduleGroup
    Properties:
      Name: B
      ...
  ScheduleGroupB:
    Type: AWS::Scheduler::ScheduleGroup
    Properties:
      Name: B
      ...
```
Am I using `Fn::ForEach` correctly? Or is this an issue with `cfn-lint` itself? Unfortunately in my situation, I cannot bypass `cfn-lint` and force the deployment of these resources.

**Comment by kddejong** [comment]
Your indentation is off.

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::LanguageExtensions'
...
Resources:
  Fn::ForEach::Loop:
    - Id
    - - A
      - B
      - C
    - ScheduleGroup${Id}:
        Type: AWS::Scheduler::ScheduleGroup
        Properties:
          Name: !Ref Id
          ...
```

**Comment by jbq2** [comment]
Hi @kddejong, thank you for noticing. I actually figured it out at work and forgot to close the issue--apologies
